packages/adapter/lib/adapter/adapter.js

```diff
@@ -1391,11 +1390,29 @@ function Adapter(options) {
         }
     };
 
+    /**
+     * sleeps the amount of time, and then fullfills the promise
+     * will not fullfill during and after adapter shutdown
+     *
+     * @param {number} timeout - timeout in milliseconds
+     * @returns {Promise<void>} promise when timeout is over
+     */
+    this.sleep = timeout => {
```

Comment:
Personally, I'd prefer to call this `wait` or `delay` instead of `sleep`. One could think `sleep` makes the adapter actually sleep, but that's not true. Other asynchronous processes will still continue.

Reply:
delay, is also the name in c#

Reply:
i'll rename and refactor